addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,899 @@
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing IRtrans Ethernet handler.");
+
+        // register ourselves as a Transceiver Status Listener
+        registerTransceiverStatusListener(this);
+
+        try {
+            selector = Selector.open();
+        } catch (IOException e) {
+            logger.error("An exception occurred while registering the selector: '{}'", e.getMessage());
+        }
+
+        if (selector != null) {
+            if (getConfig().get(IP_ADDRESS) != null && getConfig().get(PORT_NUMBER) != null) {
+                onUpdate();
+            } else {
+                logger.warn("Cannot connect to IRtrans Ethernet device. IP address or port number not set.");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+
+        unregisterTransceiverStatusListener(this);
+
+        disconnect(socketChannel);
+
+        try {
+            if (selector != null) {
+                selector.close();
+            }
+        } catch (IOException e) {
+            logger.error("An exception occurred while closing the selector: '{}'", e.getMessage());
+        }
+
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private void establishConnection() {
+        lock.lock();
+        try {
+            if (getConfig().get(IP_ADDRESS) != null && getConfig().get(PORT_NUMBER) != null) {
+                socketChannel = connect((String) getConfig().get(IP_ADDRESS),
+                        ((BigDecimal) getConfig().get(PORT_NUMBER)).intValue());
+                try {
+                    Thread.sleep(((BigDecimal) getConfig().get(RESPONSE_TIME_OUT)).intValue());
+                } catch (NumberFormatException | InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    logger.debug("An exception occurred while putting a thread to sleep: '{}'", e.getMessage());
+                }
+                onConnectable(socketChannel);
+            }
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    private synchronized void onUpdate() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0,
+                    ((BigDecimal) getConfig().get(REFRESH_INTERVAL)).intValue(), TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public void onConnectionLost() {
+        logger.debug("Updating thing status to OFFLINE.");
+        updateStatus(ThingStatus.OFFLINE);
+        for (TransceiverStatusListener listener : transceiverStatusListeners) {
+            listener.onBridgeDisconnected(this);
+        }
+        establishConnection();
```

Comment:
Wouldn't you want to wait a little bit before? E.g. schedule a reconnect in a minute or so?